### PR TITLE
Graphweaver Standalone Server Mode

### PR DESCRIPTION
**Pull request by steve-keep**
## Summary by CodeRabbit

- **New Features**
	- Added a launch configuration for "End to End - Sqlite" to streamline testing workflows in the development environment.

- **Bug Fixes**
	- Updated method calls in multiple test cases to enhance clarity and streamline code by removing unnecessary references.

- **Chores**
	- Enhanced the structure of test cases related to the `Graphweaver` class for improved readability and maintainability.

### CODE REVIEW

**Comment by steve-keep** [comment]
@coderabbitai review

File: src/packages/server/src/server.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+		}
+
+		// Order is important here
+		const plugins = [
+			LogRequests,
+			LogErrors,
+			corsPlugin(this.config.corsOptions),
+			...apolloPlugins,
+			...(this.config.graphqlDeduplicator?.enabled ? [dedupeGraphQL] : []),
+		];
```
Comment (thekevinbrown) [inline src/packages/server/src/server.ts:88]: Probably easier to read if we delete line 87 and just do:
```suggestion
		];
		
		if (this.config.graphqlDeduplicator?.enabled) plugins.push(dedupeGraphQL);
```

File: src/packages/server/src/server.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+		logger.trace(`Graphweaver buildSchemaSync finished.`);
+		logger.trace(`Graphweaver starting ApolloServer`);
+		logger.trace(`Protecting with GraphQL Armor 🛡️`);
+		const armor = new ApolloArmor(config?.graphQLArmorOptions);
+		const protection = armor.protect();
+		this.server = new ApolloServer<TContext>({
+			...(this.config.apolloServerOptions as any),
+			...protection,
+			plugins: [...plugins, ...protection.plugins],
+			schema: this.schema,
```
Comment (thekevinbrown) [inline src/packages/server/src/server.ts:124]: Yeah, if this is the only thing referencing this.schema it's probably better to just not keep a reference to it so it's not confusing.

**Review by thekevinbrown — APPROVED** [review]
Looking great! Just a couple of things to make sure we're aligned on before we pop it in.

File: src/packages/builder/src/start/backend.ts [hunk 3]
```diff
@@ -96,6 +96,20 @@ export const startBackend = async ({ host, port }: BackendStartOptions) => {
 
 	await Promise.all([checkNativeModules, buildBackend]);
 
+	const buildDir = path.join('file://', process.cwd(), `./.graphweaver/backend/index.js`);
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/packages/builder/src/start/backend.ts:99]: This is taken directly from the existing code here that is working in windows:

https://github.com/exogee-technology/graphweaver/blob/main/src/packages/cli/bin/gw-types.js#L8

Shall we update both or leave as is?

File: src/packages/core/src/request-context.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import { BaseLoader } from './base-loader';
+import { logger } from '@exogee/logger';
+
+type Context = {
+	BaseLoaders: BaseLoader;
+};
+
+export class RequestContext {
+	private static storage = new AsyncLocalStorage<RequestContext>();
+	private static counter = 1;
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/packages/core/src/request-context.ts:11]: It is used on line 13?

File: src/packages/server/src/integrations/fastify.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+};
+
+export const startStandaloneServer = async <TContext extends BaseContext>(
+	{ port, host }: StartServerOptions,
+	apollo: ApolloServer<TContext>,
+	plugins: Set<GraphweaverPlugin<void>>
+) => {
+	logger.info(`Starting standalone server on ${host ?? '::'}:${port}`);
+
+	const fastify = Fastify({});
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/packages/server/src/integrations/fastify.ts:22]: Ah yeah that can go I moved the options to the listen function 👍

File: src/packages/core/src/base-loader.ts [hunk 6]
```diff
@@ -147,39 +170,77 @@ const getBaseRelatedIdLoader = <G = unknown, D = unknown>({
...
+}
+
+const sharedBaseLoader = new BaseLoader();
+
+export const BaseLoaders = {
+	loadOne: <G = unknown, D = unknown>(options: LoadOneOptions<G>) => {
+		const baseLoader = RequestContext.getBaseLoader();
+
+		if (!baseLoader) {
+			return sharedBaseLoader.loadOne<G, D>(options);
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/packages/core/src/base-loader.ts:214]: Yeah, you're right now we are creating one for both serverless and server this should no longer be needed. Will remvoe!

File: src/packages/server/src/server.ts [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+import { pluginManager } from './plugins';
+import {
+	ExecuteOperationOptions,
+	GraphQLResponse,
+} from '@apollo/server/dist/esm/externalTypes/graphql';
+import { onRequestWrapper } from './integrations/utils';
+
+export default class Graphweaver<TContext extends BaseContext> {
+	server: ApolloServer<TContext>;
+	public schema: GraphQLSchema;
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/packages/server/src/server.ts:32]: Interesting... This is not a new addition. It is only showing as an addition as it moved from `index.ts` to `server.ts`.

File: src/packages/server/src/server.ts [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+		// Configure the plugins for Graphweaver and Apollo
+		this.graphweaverPlugins = pluginManager.getPlugins();
+		const apolloPlugins = this.config.apolloServerOptions?.plugins || [];
+
+		for (const metadata of graphweaverMetadata.entities()) {
+			if (metadata.provider?.apolloPlugins && metadata.provider.apolloPlugins.length > 0) {
+				// only push unique plugins
+				const eMetadataProviderPlugins = metadata.provider.apolloPlugins.filter(
+					(plugin) => !apolloPlugins.includes(plugin)
+				);
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/packages/server/src/server.ts:65]: I agree, however, I think we should change the approach here to follow the same `addPlugin` flow as we do with Graphweaver Plugins.

Shall we do that in this PR or do as part of the Auth AdminUI cleanup?

File: src/packages/core/src/request-context.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (thekevinbrown) [reply src/packages/core/src/request-context.ts:11]: Yup it is.

File: src/packages/server/src/server.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (thekevinbrown) [reply src/packages/server/src/server.ts:32]: This is used for types generation, which needs to know what 'the schema' is.

File: src/packages/server/src/server.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/packages/server/src/server.ts:65]: Added this task to https://exogee.atlassian.net/browse/EXOGW-348

File: src/packages/server/src/server.ts (hunk 1 above)
Comment (steve-keep) [reply src/packages/server/src/server.ts:88]: This will be updated with https://exogee.atlassian.net/browse/EXOGW-348

**Pull request merged**